subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultRepositoryContentDescriptor.java

```diff
@@ -45,6 +51,8 @@
 
     private Action<? super ArtifactResolutionDetails> cachedAction;
     private final Supplier<String> repositoryNameSupplier;
+    private final VersionSelectorScheme versionSelectorScheme = new DefaultVersionSelectorScheme(new DefaultVersionComparator(), new VersionParser());
+    private final ConcurrentHashMap<String, VersionSelector> versionSelectors = new ConcurrentHashMap<>();
```

Comment:
Since it is possible to have multiple includes/excludes with the same selectors, example: `[1.0,)`, it would be better if we re-use them